internal/gps/vcs_repo.go

```diff
@@ -138,6 +176,25 @@ type hgRepo struct {
 	*vcs.HgRepo
 }
 
+func newHgRepo(ustr, path string) (*hgRepo, error) {
+	r, err := vcs.NewHgRepo(ustr, path)
+	if err != nil {
+		// if vcs could not initialize the repo due to a local error
+		// then the local repo is in an incorrect state. Remove and
+		// treat it as a new not-yet-cloned repo.
+		if _, ok := err.(*vcs.LocalError); ok {
+			os.RemoveAll(path)
+			r, err = vcs.NewHgRepo(ustr, path)
+		}
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &hgRepo{r}, nil
+}
+
```

Comment:
I'm 👍 to @sectioneight's suggestion. There's a lot in this part of the codebase that we more or less have to duplicate, so when we come across something that we can add a little reasonable indirection to be DRY with, it's generally a good idea.